ExtraRoles/PerformKillPatch.cs

```diff
@@ -24,49 +49,88 @@ public static bool Prefix()
             if (PlayerControl.LocalPlayer.Data.IsDead)
                 return false;
 
-            MessageWriter writer;
             if (CurrentTarget != null)
             {
                 var target = CurrentTarget;
                 //code that handles the ability button presses
                 if (PlayerControl.LocalPlayer.isPlayerRole("Officer"))
                 {
-                    if (PlayerTools.getOfficerCD() != 0)
+                    if (PlayerTools.getOfficerCD() > 0)
+                    {
                         return false;
-                    
-                    var isPlayerJokerAndCanDieToOfficer =
-                        Main.Config.jokerCanDieToOfficer && target.isPlayerRole("Joker");
-                    
-                    writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
-                        (byte) CustomRPC.OfficerKill, Hazel.SendOption.None, -1);
-                    writer.Write(PlayerControl.LocalPlayer.PlayerId);
-                    writer.Write(!target.isPlayerImmortal() &&
-                        (isPlayerJokerAndCanDieToOfficer || target.Data.IsImpostor) ?
-                        target.PlayerId : PlayerControl.LocalPlayer.PlayerId); // kill target or suicide
-                    AmongUsClient.Instance.FinishRpcImmediately(writer);
-                    PlayerControl.LocalPlayer.MurderPlayer(target);
-                    PlayerControl.LocalPlayer.getModdedControl().LastAbilityTime = DateTime.UtcNow;
+                    }
 
                     // attempted to kill shielded player
                     if (target.isPlayerImmortal())
                     {
-                        writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
-                            (byte) CustomRPC.AttemptSound, Hazel.SendOption.None, -1);
-                        AmongUsClient.Instance.FinishRpcImmediately(writer);
+                        // suicide packet
+                        WriteKillRpc(PlayerControl.LocalPlayer, PlayerControl.LocalPlayer);
```

Comment:
Lots of repeating code again. Shield break should be moved to the end, kill rpc should not be here.

Reply:
The Officer should die when he attacks a shielded player. No matter what role that player is. 
![image](https://user-images.githubusercontent.com/35311793/109945789-e5ca8600-7cd7-11eb-84fc-781c3eb2f6dc.png)

Reply:
He still would have with the code I provided, it was checking if the target is not shielded before performing the kill. I did however forget to include the same check in the "CrewDie" condition, it should be there too.